### ISSUE TITLE
[a11y Documentation] Chatbot Documentation: Create and publish mobile on-device a11y test plan for Chatbot Rx and all VA.gov 

### ISSUE THREAD

**Issue by artsymartha68** [department-of-veterans-affairs/va.gov-team#73543]
> [!NOTE]
> - **This ticket is for the **Chatbot RX version** of the testing documentation.** 
>     - **This is a transition item from Eli** 
>     - Ticket Ref. #83204
>   -  Consolidating ticket and updating with notes, as of 8/22 and 8/23/2024.
> - **This ticket includes:**
>      - Background information and past guidance from July 2023. 
>      - Where Eli left off in May 2024, with his guidance via Google Docs 
> 
> @coforma-jamie @SarahKay8 @rileyorr 

# About this Ticket
_"Adding this tracking ticket for work that @eli-oat already has underway to create official documentation for the on-device mobile testing process we'd like to establish for VA.gov." - Martha c/o this ticket_

## Desired outcomes

_"This stems from the a11y work Eli is doing with the Chatbot Rx team, where we realized that we are lacking on-device testing standards and processes for this critical Veteran-facing feature." - Martha c/o this ticket_

### Goal
Testing standards and processes for chatbot, which can be used across VA, not just MHV. 

> [!CAUTION]
> **Per Eli:** Creating testing procedure for this chatbot - not sure if the team wants it in GH because it is proprietary. Team would have a copy and CAIA. Final location TBD.

### Current Resources 

**Via Google Drive:** Shared VA CAIA > a11y > Knowledge Transfer > [Chatbot Folder](https://drive.google.com/drive/folders/1eU6IrCmKTMJMaNX5J58ZA2GEt71WBPxm?usp=share_link)

**This folder includes**: 
- **Google Doc**: Web chatbot accessibility testing guidance, which includes the following sections: 
    - Context
    - State management
    - Focus management
- **Video**: [Let’s talk chatbot](https://drive.google.com/file/d/1kiU57BlVHmRWk2D-C93lEfth3-kaoNdF/view?usp=drive_link), video walkthrough.

### Past Resource
- **July 2023**: [Accessibility Chatbot Guidance (PDF) c/o AdHoc](https://adhoc.slack.com/files/U03AM3W6R41/F05HZ8NQ1V0/chatbot_location___accessibility.pdf)
- **Previously Referenced and Utilized:** [Chatbot Accessibility Playbook](https://mitre.github.io/chatbot-accessibility-playbook/)

### Information Source
_Information obtained from this intake ticket ([78614](https://github.com/department-of-veterans-affairs/va.gov-team/issues/78614))._

<details><summary>Toggle to read more background information about chatbot</summary>

### Disability Claims Status Flow

Design and build an updated **Disability Claims Status Flow** as part of the VA.gov Chatbot. 

#### Claims Flow Inclusions

**Bot can provide:** 
1. a definition of the phase, 
2. a general claim timeline, and/or 
3. details within the phase (micro-transaction)

**The bot can direct users to other VA tools** (ex. Claims Status Tool, VA.gov pages) where information can be found.

- **The Chatbot will share:** 
    - the number of compensation claims on file
    - the most recent claim opened
    - the date it was opened
    - the related issues, and the status
- **The Chatbot will share details** on the step of the claim process that disability compensation claim is in including the current status of the claim: 
    - claim received
    - initial review
    - evidence gathering review decision
    - preparation for notification
    - complete
- **The Chatbot will share** when:
    - the claim moved status
    - the date the status changed
- **The Chatbot will provide** recent updates for the veteran to learn about the last activity done to the compensation claim. The chatbot will share: 
    - the activity
    - date submitted
    - any actions needed from the veteran
- **The Chatbot will provide** the veteran with the average number of days it takes to complete a disability compensation claim (provided by VA.gov) as of the most relevant date
- **The Chatbot will provide** the veteran with a phone number (1-800-827-1000) to click-to-call to answer questions related to disability claims status that the chatbot cannot answer

#### Current State/Abilities of the Chatbot

At this time, this chatbot can only show a Veteran status information on disability compensation claims. The chatbot will provide the date the claim was filed and the current phase that the compensation claim is in. 

#### Items Found in the Claims Status Tool (Outside of Chatbot)

All other details of the in-progress claims and other claims must be found via the Claims Status Tool or through an agent. 

#### Seamless Experience Desired

To provide a seamless experience across VA platforms and channels, the Chatbot should have the same capabilities in this area as other tools on the VA platform, such as the Claims Status Tool. 

**_Responding to select claims status details and not providing details to the veteran could harm the veteran's confidence in the bot over the long term._**
</details>

## Publishing Location 

**Location TBD**

_"It would be good to have this in Confluence for collaboration to begin. Then I could see it going onto the Platform website, design.va.gov, other options TBD?" - Martha c/o this ticket_

### Location Options & Considerations
_See cautionary note from Eli above._ 
- [ ] GitHub Documentation - CAIA 
    - **Note**: _Current testing documentation for desktop, mobile and Figma, lives here._
    - Eli started a folder here: `teams/CAIA/accessibility/Deliverables/VA Chatbot`
        - `WIP`: [chatbot-accessibility-guidance-and-resources.md](https://github.com/department-of-veterans-affairs/va.gov-team/blob/e54350d9fd0468a470f804634433ad51033bfb92/teams/CAIA/accessibility/Deliverables/VA%20Chatbot/chatbot-accessibility-guidance-and-resources.md)
- [ ] Platform Website
- [ ] Design System Website
    - **Note**: _Discussed with Eli, prior to departure. VADS section for Chatbots need._
- [ ] Confluence (limited access to a11ys only on CAIA - at the moment)

## Collaboration timeframe

_"March 2024 would be great!" - Martha c/o original ticket_ 

### Original Timeline Adjusted
- There has been a delay, as Eli's last day was May 8, 2024. 
### Future CAIA Chatbot Testing Work
- @SarahKay8 (CAIA) has taken lead on the VA Chatbot project, and may be able to assist with this. 
- Additionally, @coforma-jamie (CAIA) has worked on the mobile testing documentation with @rileyorr (CAIA/MHV) and may be able to assist as well. 

## CAIA & A11y Tasks

<br>

```[tasklist]
### CAIA operations
- [x] Initial Review - @coforma-terry and @strelichl
- [x] Ticket refinement - Sara 8/22 and 823/2024
```

```[tasklist]
### Mobile Testing Guidance
- [x] 2/21/2024 @elioat completed draft mobile accessibility guidance
- [x] 2/21/2024 Sharing with Martha and Danielle
- [x] 2/21/2024 @elioat [Shared in a11y specialists channel in DSVA Slack](https://dsva.slack.com/archives/C0335Q88X0S/p1708533625955899) - the wider community doing accessibility at OCTO
- [x] 3/5/2024 Sharing the OCTO community via the accessibility-help channel
- [x] [Pilot documentation for mobile web accessibility](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/CAIA/accessibility/mobile-accessibility-testing-guidance.md)
- [x] Documentation refinement, based on requested feedback
- [ ] #90459
```

```[tasklist]
### Chatbot Documentation Guidance
- [x] **DRAFT** documentation for Chatbots
- [x] **Review** [existing Chatbot Guidance from July 2023 (AdHoc) PDF](https://adhoc.slack.com/files/U03AM3W6R41/F05HZ8NQ1V0/chatbot_location___accessibility.pdf?origin_team=T1ZD0UBMZ&origin_channel=C01K23NSRGX)
- [x] **Reference Item:** [Additional Starting Point Chatbot Accessibility Playbook](https://mitre.github.io/chatbot-accessibility-playbook/docs/A_2.html)
- [x] **Review** [guidance started by Eli in Google Docs](https://drive.google.com/drive/folders/1eU6IrCmKTMJMaNX5J58ZA2GEt71WBPxm?usp=share_link)
- [x] **Review Background Information section found in this ticket:** #79014
- [x] **REVIEW** documentation for Chatbots with CAIA Team (and a11y community of practice?)
- [x] **FINALIZE:** documentation for Chatbots
- [ ] **DETERMINE:** Home for final documentation (see cautionary statement above in ticket and locations section). (Waiting on Anita's feedback)
- [ ] Create demos
- [ ] Create trainings
```

```[tasklist]
### Related
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/79014
```

**Comment by coforma-jamie** [comment]
The [Chatbot Accessibility Playbook](https://mitre.github.io/chatbot-accessibility-playbook/) is probably the best resource to develop a testing plan from. I believe we found this early on and used it to help us evaluate the Rx skill.

**Comment by coforma-jamie** [comment]
[Web Chat Accessibility Considerations](https://www.craigabbott.co.uk/blog/web-chat-accessibility-considerations/) is another great resource. Spells things out in plain language and associates them with WCAG criteria

**Comment by SarahKay8** [comment]
Confirmed with Chatbot team this is good to close out

CC: @anitadewitt-va @artsymartha68

**Comment by SarahKay8** [comment]
Confirmed with Chatbot team this is good to close out

CC: @anitadewitt-va @artsymartha68 @NaomiPMC @sara-amanda

**Comment by sara-amanda** [comment]
[Per SK via Slack](https://oddball.slack.com/archives/C05LB36ATAQ/p1727986457072969?thread_ts=1727701900.927949&cid=C05LB36ATAQ): 
> Anita said to close 78614 and 73543

**Comment by sara-amanda** [comment]
> ### Goal
> 
> Testing standards and processes for chatbot, which can be used across VA, not just MHV.

> [!CAUTION]
> 
> Per Eli: Creating testing procedure for this chatbot - not sure if the team wants it in GH because it is proprietary. Team would have a copy and CAIA. Final location TBD.

## Update: Location for Documentation

- [x] Moving ticket to `Validate` 
- [ ] We will note final documentation location, when confirmed, prior to closing this out officially. 
- [ ] Create demos and trainings `TBD - Separate Ticket?`

**Comment by sara-amanda** [comment]
@NaomiPMC Per SK, we are going to wait for Anita to provide feedback. See details above.

**Comment by sara-amanda** [comment]
[Moving to Sprint 138](https://github.com/department-of-veterans-affairs/va.gov-team/issues/73543#issuecomment-2392272715)

**Comment by mistymg** [comment]
@lakper This is still showing up on the Content and IA board, so I'm going to remove it from ours :)
